app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java

```diff
@@ -244,16 +244,32 @@ private void getDisplayModes() {
     }
 
     @TargetApi(23)
-    private Display.Mode findBestDisplayMode(Float refreshRate) {
-        if (mDisplayModes == null || refreshRate == null) return null;
-
-        int sourceRate = Math.round(refreshRate);
-        for (Display.Mode mode : mDisplayModes){
-            int rate = Math.round(mode.getRefreshRate());
-            if (rate == sourceRate || rate == sourceRate * 2) return mode;
+    private Display.Mode findBestDisplayMode(MediaStream videoStream) {
+        if (mDisplayModes == null || videoStream.getRealFrameRate() == null) return null;
+
+        int curWeight = 0;
+        Display.Mode bestMode = null;
+        int sourceRate = Math.round(videoStream.getRealFrameRate() * 100);
+        for (Display.Mode mode : mDisplayModes) {
+            // Skip unwanted display modes
+            if (mode.getPhysicalWidth() < 1280 || mode.getPhysicalHeight() < 720)  // Skip non-HD
+                continue;
```

Comment:
Does it make sense to disallow low resolution? What if my video is 480p? The next if statement already disallows display resolutions lower then the video resolution.

Reply:
Not per-se mandatory, but in my experience, things get murky when going to non-HD resolutions, so this is just to keep it safe.